src/dev/pca9702.rs

```diff
@@ -0,0 +1,194 @@
+    fn is_set(&mut self, _mask_high: u32, _mask_low: u32) -> Result<u32, Self::Error> {
+        Err(Pca9702Error::OutputNotSupported)
+    }
+
+    /// Read the actual input bits from the PCA9702 device
+    fn get(&mut self, mask_high: u32, mask_low: u32) -> Result<u32, Self::Error> {
+        let val = self.bus.read_inputs().map_err(Pca9702Error::Bus)? as u32;
+        Ok((val & mask_high) | (!val & mask_low))
+    }
+}
+
+/// Bus wrapper type for PCA9702, implementing `Pca9702BusTrait`.
+pub struct Pca9702Bus<SPI>(pub SPI);
+
+impl<SPI> Pca9702BusTrait for Pca9702Bus<SPI>
+where
+    SPI: SpiBus,
+{
+    type BusError = SPI::BusError;
+
+    fn read_inputs(&mut self) -> Result<u8, Self::BusError> {
+        // PCA9702 wants a total of 8 SCLK rising edges to shift out the input data
+        // from SDOUT: The first rising edge latches the inputs, the next 8 edges
+        // shift them out.
+        let mut buffer = [0u8];
+        let mut ops = [Operation::TransferInPlace(&mut buffer)];
+        self.0.transaction(&mut ops)?;
+
+        // buffer[0] now holds bits [in7..in0]
+        Ok(buffer[0])
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use embedded_hal_mock::eh1::spi::{Mock as SpiMock, Transaction as SpiTransaction};
+
+    #[test]
+    fn pca9702_read_inputs() {
+        let expectations = [
+            // 1st read
+            SpiTransaction::transaction_start(),
+            SpiTransaction::transfer_in_place(vec![0], vec![0xA5]),
+            SpiTransaction::transaction_end(),
+            // 2nd read
+            SpiTransaction::transaction_start(),
+            SpiTransaction::transfer_in_place(vec![0], vec![0xA5]),
+            SpiTransaction::transaction_end(),
+            // 3rd read
+            SpiTransaction::transaction_start(),
+            SpiTransaction::transfer_in_place(vec![0], vec![0xA5]),
+            SpiTransaction::transaction_end(),
+        ];
+        let spi_mock = SpiMock::new(&expectations);
+        let mut pca = Pca9702::new(spi_mock);
+        let pins = pca.split();
+
+        // For each call, the driver re-reads from SPI, returning 0xA5 each time.
+        // 0xA5 = 0b10100101 => in0=1, in1=0, in2=1, in3=0, in4=0, in5=1, in6=0, in7=1
+        assert_eq!(pins.in0.is_high().unwrap(), true); // LSB => 1
+        assert_eq!(pins.in1.is_high().unwrap(), false);
+        assert_eq!(pins.in2.is_high().unwrap(), true);
+
+        // Finally, consume the mock:
+        let mut spi = pca.0.into_inner().bus.0;
+        spi.done();
+    }
+
+    #[test]
+    fn pca9702_output_fails() {
+        let spi_mock = SpiMock::new(&[]);
+        let pca = Pca9702::new(spi_mock);
+
+        let err = pca.0.borrow_mut().set(0x01, 0).unwrap_err();
+        match err {
+            Pca9702Error::OutputNotSupported => {}
+            _ => panic!("Expected OutputNotSupported error"),
+        }
+        let mut spi = pca.0.into_inner().bus.0;
+        spi.done();
+    }
```

Comment:
This then needs to become a `#[should_panic]` testcase.

Also, `pca.0` is not public API so this isn't something a user could do.  However, there is still a way to get there, by doing this:

```rust
pins.in0.access_port_driver(|drv| {
    drv.set(0x01, 0x00).unwrap_err();
});
```

So I'd prefer to use that accessor in the testcase as well.